// knjige/knjige.js
let knjige = [
    {
        id: 1,
        naziv: "Orlovi rano lete",
        autor: "Branko Copic"
    },
    {
        id: 2,
        naziv: "Na Zapadu nista novo",
        autor: "Erih Marija Remark"
    },
    {
        id: 3,
        naziv: "Travnicka hronika",
        autor: "Ivo Andric"       
    },
    {
        id: 4,
        naziv: "Ana Karenjina",
        autor: "Lav Tolstoj"       
    },
    {
        id: 5,
        naziv: "Cica Tomina koliba",
        autor: "Mark Tven"       
    },
]




exports.sveKnjige = () => {
    for(let i = 0; i < knjige.length;i++)
        {
            console.log(knjige[i].id + "-" + knjige[i].naziv + "-" + knjige[i].autor)
        }
}

exports.addKnjiga = (Id,naziv,Autor) => {
     knjige.push({id: parseInt(Id), naziv: naziv, autor: Autor})
}

exports.getKnjiga = (id) => {
    for(let i = 0; i < knjige.length;i++)
        {
            if(id == knjige[i].id)
                {
                     console.log(knjige[i].id + "-" + knjige[i].naziv + "-" + knjige[i].autor)
                }              
        }
}

exports.postaviAutora = (naziv,Autor) => {
    for(let i = 0; i < knjige.length;i++)
        {
            if(naziv == knjige[i].naziv)
                {
                    knjige[i].autor = Autor
                }              
        }
}

exports.deleteKnjiga = (Id) => {
    for(let i = 0; i < knjige.length;i++)
        {
            if(parseInt(Id) == knjige[i].id)
                {
                    knjige.pop(i)
                }          
        }
}

exports.getKnjigaByAuthor = (author) => {
    for(let i = 0; i < knjige.length;i++)
        {
            if(author == knjige[i].autor)
                {
                    console.log(knjige[i].id + "-" + knjige[i].naziv)
                }              
        }
}
